pkg/sgtm/page_settings.go

```diff
@@ -38,25 +40,27 @@ func (svc *Service) settingsPage(box *packr.Box) func(w http.ResponseWriter, r *
 				soundcloud = strings.TrimPrefix(soundcloud, "https://soundcloud.com/")
 				soundcloud = strings.TrimPrefix(soundcloud, "@")
 
-				fields := map[string]interface{}{
-					"firstname":           strings.TrimSpace(r.Form.Get("firstname")),
-					"lastname":            strings.TrimSpace(r.Form.Get("lastname")),
-					"homepage":            strings.TrimSpace(r.Form.Get("homepage")),
-					"bio":                 strings.TrimSpace(r.Form.Get("bio")),
-					"headline":            strings.TrimSpace(r.Form.Get("headline")),
-					"inspirations":        strings.TrimSpace(r.Form.Get("inspirations")),
-					"gears":               strings.TrimSpace(r.Form.Get("gears")),
-					"goals":               strings.TrimSpace(r.Form.Get("goals")),
-					"genres":              strings.TrimSpace(r.Form.Get("genres")),
-					"other_links":         strings.TrimSpace(r.Form.Get("other_links")),
-					"twitter_username":    twitter,
-					"soundcloud_username": soundcloud,
+				user := &sgtmpb.User{
+					Firstname:          strings.TrimSpace(r.Form.Get("firstname")),
+					Lastname:           strings.TrimSpace(r.Form.Get("lastname")),
+					Homepage:           strings.TrimSpace(r.Form.Get("homepage")),
+					Bio:                strings.TrimSpace(r.Form.Get("bio")),
+					Headline:           strings.TrimSpace(r.Form.Get("headline")),
+					Inspirations:       strings.TrimSpace(r.Form.Get("inspirations")),
+					Gears:              strings.TrimSpace(r.Form.Get("gears")),
+					Goals:              strings.TrimSpace(r.Form.Get("goals")),
+					Genres:             strings.TrimSpace(r.Form.Get("genres")),
+					OtherLinks:         strings.TrimSpace(r.Form.Get("other_links")),
+					TwitterUsername:    twitter,
+					SoundcloudUsername: soundcloud,
 				}
-				return fields
+
+				return user
 			}
 			fields := validate()
 			if fields != nil {
-				if err := svc.rwdb().Model(data.User).Updates(fields).Error; err != nil {
+				err := svc.storage.UpdateUser(fields, fields)
```

Comment:
```suggestion
				err := svc.storage.UpdateUser(data.User, fields)
```

because `fields` does not contain an ID